wynnset/tapestry-wp
# Add new locking conditions based on date

Closes #484

* Adds two new conditions that allow authors to lock/unlock nodes based on current date
* Update authoring tool to show more descriptive conditions

**Testing**

* Create a new node with the new conditions
* Set a date with the date picker
* The node should be locked/unlocked according to the date that is set and the chosen condition

**Note**

This is implemented in a way that allows it to be completed quickly, that is I hard coded a `date` property into the condition object.

In the future, I think we should change the condition object itself to contain a `data` property instead of separate `nodeId` and `date` properties. This allows the condition to be flexible to whatever future condition we want the node to rely on. Since a lot of the code right now expects a `nodeId` at the top level of the condition object, this requires some refactoring in that regard.

For example, if we want to unlock a node based on the current tapestry progress, all we need is to add attributes to the `data` property rather than hard code these new values in.

**Todo**

- [x] Add support for time and timezone
